tests/unit/test_fileio.py

```diff
@@ -64,22 +71,22 @@ def read_from_fake_data(start=0, end=None, **_):
         self.assertEqual(reader._pos, 12)
         self.assertEqual(blob.download_as_bytes.call_count, 2)
         blob.download_as_bytes.assert_called_with(
-            start=8, end=16, checksum=None, **download_kwargs
+            start=8, end=16, checksum=None, retry=DEFAULT_RETRY, **download_kwargs
         )
 
         # Read a larger amount, requiring a download larger than chunk_size.
         self.assertEqual(reader.read(16), TEST_BINARY_DATA[12:28])
         self.assertEqual(reader._pos, 28)
         self.assertEqual(blob.download_as_bytes.call_count, 3)
         blob.download_as_bytes.assert_called_with(
-            start=16, end=28, checksum=None, **download_kwargs
+            start=16, end=28, checksum=None, retry=DEFAULT_RETRY, **download_kwargs
         )
 
         # Read all remaining data.
         self.assertEqual(reader.read(), TEST_BINARY_DATA[28:])
         self.assertEqual(blob.download_as_bytes.call_count, 4)
         blob.download_as_bytes.assert_called_with(
-            start=28, end=None, checksum=None, **download_kwargs
+            start=28, end=None, checksum=None, retry=DEFAULT_RETRY, **download_kwargs
         )
 
         reader.close()
```

Comment:
Do we need a testcase which passes a non-default `retry` to reader, and verifies that it is passed through?

Reply:
Added as `test_retry_passed_through`